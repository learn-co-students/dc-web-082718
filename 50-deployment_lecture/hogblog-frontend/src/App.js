import React, { Component } from "react";
import pig from "./pig.png";
import "./App.css";

import PigContainer from "./PigContainer";

class App extends Component {
  state = {
    pigs: []
  };
  componentDidMount() {
    // console.log(process.env);
    const URL = `https://hogblog-backend.herokuapp.com/hogs`;
    fetch(URL)
      .then(res => res.json())
      .then(pigs =>
        this.setState({
          pigs
        })
      );
  }
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={pig} className="App-logo" alt="logo" />
          <h1>Welcome to hog.blog</h1>
          <PigContainer pigs={this.state.pigs} />
        </header>
      </div>
    );
  }
}

export default App;
