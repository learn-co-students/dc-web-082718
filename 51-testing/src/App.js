import React, { Component } from "react";
import { connect } from "react-redux";
import "./App.css";

import AddTodo from "./components/AddTodo";
import Todos from "./components/Todos";
import { loadData } from "./redux/actions";

class App extends Component {
  // dispatches the action that sends the 'mockData' object
  // through the reducer and to the store
  componentDidMount() {
    this.props.loadData();
  }

  // WHY redux? Just look how neat and tidy the App component is!
  render() {
    return (
      <div className="App">
        <br />
        <h1 className="ui header">
          <i className="gem icon" />
          To Do
        </h1>
        <Todos />
        <AddTodo />
      </div>
    );
  }
}

// maps the 'loadData' action creator to 'App.js' props. When executed, this
// kicks off the cycle from 'action' => 'reducer' => 'update state'.
// (i.e. it dispatches the action created BY the loadData function)
const mapDispatchToProps = dispatch => {
  return {
    loadData: () => dispatch(loadData())
  };
};

// connecting our dispatched loadData function to App's props.
export default connect(
  null,
  mapDispatchToProps
)(App);

export { App };
