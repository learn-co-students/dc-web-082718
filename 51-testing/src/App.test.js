import React from "react";
import ReactDOM from "react-dom";
import { App } from "./App";
import { shallow } from "enzyme";

it("renders without crashing", () => {
  let loadData = () => "test data";
  let wrapper = shallow(<App loadData={loadData} />);
  expect(wrapper).toMatchSnapshot();
});
