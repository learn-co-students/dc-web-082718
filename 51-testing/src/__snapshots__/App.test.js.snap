// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="App"
>
  <br />
  <h1
    className="ui header"
  >
    <i
      className="gem icon"
    />
    To Do
  </h1>
  <Connect(Todos) />
  <Connect(AddTodo) />
</div>
`;
